lrlex/src/lib/parser.rs

```diff
@@ -104,11 +154,15 @@ impl<StorageT: TryFrom<usize>> LexParser<StorageT> {
             if let Some(j) = self.lookahead_is("%%", i) {
                 break Ok(j);
             }
-            i = self.parse_declaration(i)?;
+            i = self.parse_declaration(i, errs)?;
```

Comment:
Do we not want to gather all declaration errors into `errs`, rather than aborting on the first failure?
Or is the intent to get the initial code refactoring into place, then gradually widen the permitted collection?

Reply:
It really depends upon the error, there is primarily two factors
1. whether we can continue parsing after the error occurs at all.
2. if the continuing to parse will cause a cascade of subsequent errors if we ignore it

All of the "Duplicated" errors, tend to more obviously fall in the category of being able to continue parsing without causing a cascade of errors, since we can just not add the duplicate state.

Other errors are likely to require some experimentation to see how they end up reacting.
`UnknownDeclaration` seems like it could probably be ignored, I'm less certain how `InvalidStartStateName`, since it is likely to occur because of a syntax error during e.g. an incomplete edit.

So yeah, the intent is to just get the initial obvious cases working, enough so that it is tested, and probably fixing more cases as we uncover them.

Reply:
Sounds good. I'm happy with that as a plan for this first refactor. Once in, much easier for multiple people to pick up individual pieces separately and work through the failure modes, and consequences of continuing after each possible error.